daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DataProcessor.scala

```diff
@@ -685,6 +685,8 @@ class ParseResult(dp: DataProcessor, override val resultState: PState)
   extends DFDL.ParseResult
   with WithDiagnosticsImpl {
 
+  private var valres: Option[ValidationResult] = None
```

Comment:
Yeah, Id thought about that sort of change too, not a fan of the var, but it seemed like it would grow the scope of this PR more than I liked so avoided it.

Will take a quick look to see how much it would involve to do that refactor.  I wouldnt mind doing it.

Reply:
The Scala and Java APIs got updates to the ParseResult to expose the validation result.  What other updates were you looking for?

Reply:
The ParseResult refactor went well